data/src/client.rs

```diff
@@ -2998,6 +3002,10 @@ impl Client {
     }
 
     pub fn tick(&mut self, now: Instant) -> Result<()> {
+        if let Some(ref mut anti_flood) = self.anti_flood {
```

Comment:
I am not an expert in Rust, but putting anti_flood inside a Some seems a bit unecessary. You could probably use `matches!` here: `if matches!(self.anti_flood, Some(_)) {`


(I could be wrong here though)